src/firefly/test/edu/caltech/ipac/astro/IpacTableReaderTest.java

```diff
@@ -0,0 +1,801 @@
+package edu.caltech.ipac.astro;
+
+
+import edu.caltech.ipac.firefly.util.FileLoader;
+import edu.caltech.ipac.util.DataGroup;
+import edu.caltech.ipac.util.DataObject;
+import edu.caltech.ipac.util.DataType;
+import edu.caltech.ipac.util.StringUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.Reader;
+import java.util.*;
+
+/**
+ * Created by ymei on 1/26/17.
+ *
+ * DM-8916-JUnitTestIPACTableRead
+ *
+ * This test is mainly test if IPACTableReader.java reads the IPAC table correctly. The IRSA IPAC table reader requirements are in
+ *      http://irsa.ipac.caltech.edu/applications/DDGEN/Doc/ipac_tbl.html
+ *
+ * An IPAC table is composed by three parts: Attributes which includes keywords and comments; header which includes column names,
+ *      data types, data units, and null values; and data rows.
+ *
+ * The test cases below test if the reader can parse all parts in a table correctly and also if the reader can throw exceptions correctly
+ *      when a table violate the requirements.
+ *
+ */
+public class IpacTableReaderTest {
+
+    //Input parameters:
+    private static final String catName = "catName";
+    String[] onlyColumns = null;
+    private boolean useFloatsForDoubles = false;
+    private boolean isHeadersOnlyAllow = false;
+    private boolean noAttributes = false;
+    private int estFileSize = 0;
+
+    //Expected parameters in the data group:
+    private String[] attributeKeys = null;
+    private String[] attributeValues = null;
+    private String[] attributeComments = null;
+    private String[] colTitles = null;
+    private String[] colKeyNames = null;
+    private String[] colDataTypeDesc = null;
+    private String[] colDataTypes = null;
+    private String[] colDataUnits = null;
+    private String[] colNullStrings = null;
+    private String[][] dataValues = null;
+
```

Comment:
The best practice is to build an expected DataGroup in your setUp method, and then write it to a temporary file. Use this expected DataGroup to compare with the DataGroup read in from the file by IpacTableReader.